### PR TITLE
Dev 200

### PR DESCRIPTION
**Pull request by Palabola**
## Summary by CodeRabbit

- **New Features**
  - Added JSON-LD schema generation for server details, enhancing SEO and search engine visibility.
  - Implemented cleaning up of structured data upon component destruction to maintain data integrity.

- **Bug Fixes**
  - Correctly initialized JSON object in articles to ensure proper SEO structured data setup.
  - Fixed thumbnail restoration in SEO handler to use the correct variable for meta tag updates.

### CODE REVIEW

File: src/app/pages/server-details/server-details.component.ts [hunk 1]
```diff
@@ -1159,4 +1167,41 @@ export class ServerDetailsComponent implements OnInit {
...
+      "brand": {
+        "@type": "Organization",
+        "name": this.serverDetails.vendor.name,
+        "logo": this.serverDetails.vendor.logo,
+        "url": this.serverDetails.vendor.homepage
+      }
+    }
+
+    if(this.similarByFamily.length) {
+      json['isSimilarTo'] = this.similarByPerformance.map((s) => {
```
Comment (daroczig) [inline src/app/pages/server-details/server-details.component.ts:1195]: length check is done for `similarByFamily` but using `similarByPerformance` looks odd

File: src/app/pages/server-details/server-details.component.ts [hunk 2]
```diff
@@ -1159,4 +1167,41 @@ export class ServerDetailsComponent implements OnInit {
...
+        return {
+          "@type": "Product",
+          "name": s.display_name,
+          "url": `https://www.sparecores.com/server/${s.vendor_id}/${s.api_reference}`
+        }
+      });
+    }
+
```
Comment (daroczig) [inline src/app/pages/server-details/server-details.component.ts:1203]: it seems that we need to provide pricing as well:

image

I think offers/price is what we can do (including the lowest) as per https://developers.google.com/search/docs/appearance/structured-data/product-snippet#structured-data-type-definitions

File: src/app/pages/server-details/server-details.component.ts [hunk 3]
```diff
@@ -1159,4 +1167,41 @@ export class ServerDetailsComponent implements OnInit {
...
+  generateSchemaJSON(title: string, description: string, keywords: string) {
+    if(!this.serverDetails) {
+      return;
+    }
+
+    console.log('Generating JSON-LD schema for server details page', this.serverDetails);
+
+    const json: any =
+    // generate schema.org Product JSON-LD
+    //const server = this.serverDetails;
```
Comment (daroczig) [inline src/app/pages/server-details/server-details.component.ts:1180]: cleanup, pls

Comment (daroczig) [reply src/app/pages/server-details/server-details.component.ts:1180]: same for above `console.log`

File: src/app/pages/server-details/server-details.component.ts (hunk 1 above)
Comment (Palabola) [reply src/app/pages/server-details/server-details.component.ts:1195]: oops sry

**Review by daroczig — APPROVED** [review]

**Pull request merged**